### DISCUSSION TITLE
Jinja templating for Groupby fields

### DISCUSSION THREAD

**Discussion by prb219** [apache/superset#14396] [Q&A / Help]
**Describe the solution you'd like**
Hi, I am trying to create a filter in my dashboard to choose various groupby fields to be applied to my charts. My approach to this was to try using jinja templating in my charts to pick the value chosen in the filter to be applied as groupby field. However, I do not see a 'custom sql' option to provide jinja template to a group by field. Can you please suggest me a workaround to solve this usecase ? 

**Describe alternatives you've considered**
As an alternative I must be able to use jinja templating in my datasource itself but then it might not be the best idea. 

Superset version in use is 0.37.0

**Comment by nytai** [comment]
have you tried using a calculated column in the dataset definition? "Edit dataset" in explore

**Reply by prb219** [reply]
@nytai I tried adding a calculated column in the dataset but my jinja template is not getting replaced by the filter value when i run the chart. 

Below is the sql expression i added for my calculated column: 
{{filter_values("REGION")[0]}} 

Do you have any suggestion on how i could overcome this ?

**Comment by markostefanovic1** [comment]
Hello, I'm interested in this as well. An explanation for how to make a dummy column that will use filter_values() jinja variable can be found in [this video.](https://www.youtube.com/watch?v=1QpG_m_ix3w) 

But this is not the real solution - what if we want to have multiple columns in the GROUP BY statement? 

```SQL 
GROUP BY
    column_1,
    column_2
```

In this case we can't have 1 dummy column for 2 GROUP BY values because Explore UI will not update the columns applied to the Dimnsions field in Superset table (aggregated metrics) or in the Columns field (raw records). 
![Screenshot 2022-05-18 at 08 03 07](https://user-images.githubusercontent.com/105298890/168968499-31f6b8e2-526e-46ee-993a-95b05ef0a568.png)
![Screenshot 2022-05-18 at 08 02 31](https://user-images.githubusercontent.com/105298890/168968509-53d2dda9-bfcf-467a-b574-fc6c2c158755.png)



Is there a way we can use Jinja templating to GROUP BY any number of columns (as many as chosen in the multi-select filter)?

Can Jinja variables "groupby" or "table_columns" be used for something like this and can someone explain the syntax for using those variables?
